### PR TITLE
Bug 2033784: set kube-apiserver degraded=true if webhook matches a virtual resource

### PR DESCRIPTION
**Pull request by sanchezl**
Add new condition: `VirtualResourceAdmissionDegraded`
Add new reason: `AdmissionWebhookMatches`

### CODE REVIEW

File: pkg/operator/webhooksupportabilitycontroller/conditions.go [hunk 1]
```diff
@@ -25,4 +29,8 @@ const (
 	// WebhookServiceNotReadyReason indicates that webhook services are having a variety of
 	// problems.
 	WebhookServiceNotReadyReason = "WebhookServiceNotReady"
+
+	// DynamicAdmissionWebhookMatchesReason indicates that a dynamic admission webhook matches
+	// a virtual resource.
+	DynamicAdmissionWebhookMatchesReason = "DynamicAdmissionWebhookMatches"
```
Comment (sttts) [inline pkg/operator/webhooksupportabilitycontroller/conditions.go:35]: not sure I would understand this in a real cluster. would drop the "dynamic". VirtualResourceAdmissionWebHook

Comment (sanchezl) [reply pkg/operator/webhooksupportabilitycontroller/conditions.go]: Done.

File: pkg/operator/webhooksupportabilitycontroller/degraded_webhook_on_virtual_resource.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		}
+	}
+
+	if len(msgs) > 0 {
+		sort.Strings(msgs)
+		condition.Message = strings.Join(msgs, "\n")
+		condition.Reason = AdmissionWebhookMatchesVirtualResourceReason
+		condition.Status = operatorv1.ConditionTrue
+	} else {
+		condition.Status = operatorv1.ConditionFalse
```
Comment (benluddy) [inline pkg/operator/webhooksupportabilitycontroller/degraded_webhook_on_virtual_resource.go:75]: Does `condition.Message` need to be cleared here?

**Pull request merged**